packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart

```diff
@@ -144,11 +144,16 @@ abstract class GoogleMapsFlutterPlatform extends PlatformInterface {
 
   /// Starts an animated change of the map camera position.
   ///
+  /// The [animationDuration] parameter indicates how long the animation should
+  /// last. If [animationDuration] is null the platform default animation
+  /// duration will be used.
+  ///
   /// The returned [Future] completes after the change has been started on the
   /// platform side.
   Future<void> animateCamera(
     CameraUpdate cameraUpdate, {
     required int mapId,
+    Duration? animationDuration,
```

Comment:
This is a breaking change; see https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-platform-interface-method-parameters for the current preferred way to handle this.

Reply:
Is this an appropriate solution ?:
```dart
  /// Starts an animated change of the map camera position.
  ///
  /// Default platform animation duration is used.
  ///
  /// The returned [Future] completes after the change has been started on the
  /// platform side.
  Future<void> animateCamera(
    CameraUpdate cameraUpdate, {
    required int mapId,
  }) {
    throw UnimplementedError('animateCamera() has not been implemented.');
  }

  /// Starts an animated change of the map camera position.
  ///
  /// The [animationDuration] parameter indicates how long the animation should
  /// last.
  ///
  /// The returned [Future] completes after the change has been started on the
  /// platform side.
  Future<void> animateCameraWithCustomDuration(
    CameraUpdate cameraUpdate, {
    required int mapId,
    required Duration animationDuration,
  }) async {
    await animateCamera(cameraUpdate, mapId: mapId);
  }
```

I didn't add a comment to specify the first one is deprecated as with this both method have their utility.